tmt/steps/__init__.py

```diff
@@ -538,76 +684,174 @@ def methods(cls) -> List[Method]:
-        return default
+
+        value = self._data_class.default(tmt.utils.option_to_key(option), default=default)
+
+        if value is None:
+            return default
+
+        return value
 
     def get(self, option: str, default: Optional[Any] = None) -> Any:
         """ Get option from plugin data, user/system config or defaults """
+
         # Check plugin data first
+        #
+        # Since self.data is a dataclass instance, the option would probably exist.
+        # As long as there's no typo in name, it would be defined. Which complicates
+        # the handling of "default" as in "return *this* when attribute is unset".
+        key = tmt.utils.option_to_key(option)
+
         try:
-            # FIXME Enable type check once StepData defined more precisely
-            return self.data[option]  # type: ignore
-        except KeyError:
-            pass
+            value = getattr(self.data, key)
+
+            # If the value is no longer the default one, return the value. If it
+            # still matches the default value, instead of returning the default
+            # value right away, call `self.default()` so the plugin has chance to
+            # catch calls for computed or virtual keys, keys that don't exist as
+            # atributes of our step data.
+            #
+            # One way would be to subclass step's base plugin class' step data class
+            # (which is a subclass of `StepData` and `SerializedContainer`), and
+            # override its `default()` method to handle these keys. But, plugins often
+            # are pretty happy with the base data class, many don't need their own
+            # step data class, and plugin developer might be forced to create a subclass
+            # just for this single method override.
+            #
+            # Instead, keep plugin's `default()` around - everyone can use it to get
+            # default value for a given option/key, and plugins can override it as needed
+            # (they will always subclass step's base plugin class anyway!). Default
+            # implementation would delegate to step data `default()`, and everyone's
+            # happy.
+
+            if value != self.data.default(key):
+                return value
 
-        # Check user config and system config
-        # TODO
+        except AttributeError:
+            pass
 
-        # Finally check plugin defaults
         return self.default(option, default)
 
     def show(self, keys: Optional[List[str]] = None) -> None:
         """ Show plugin details for given or all available keys """
+        # Avoid circular imports
+        import tmt.base
+
         # Show empty config with default method only in verbose mode
-        if (set(self.data.keys()) == set(['how', 'name'])
+        if (self.data.is_bare
                 and not self.opt('verbose')
-                and self.data['how'] == self.how):
+                and self.data.how == self.how):
```

Comment:
Isn't the `self.data.how == self.how` check is already covered by `self.data.is_bare`?

Reply:
Not necessarily, since `how` has no default value, therefore `how: 'foo'` is the same as `how: 'bar'` from the point of view of `is_bare`.

IIUIC, we might not need this anymore, but for a different reason: because `self.data.how` and `self.how` should now be aligned all the time, thanks to `delegate()` & co. not letting incompatible `how` sneak into the plugin's `data`. This check should be always true, if I'm not mistaken.

Reply:
Fixed in [8d096fd](https://github.com/teemtee/tmt/pull/1461/commits/8d096fd6dedfb1f4dab9cea49f75946075af5836)